apps/language_server/test/providers/execute_command/manipulate_pipes_test.exs

```diff
@@ -393,6 +393,78 @@ defmodule ElixirLS.LanguageServer.Providers.ExecuteCommand.ManipulatePipesTest d
                expected_substitution
              ) == edited_text
     end
+
+    test "can handle multiple calls" do
+      {:ok, _} =
+        JsonRpcMock.start_link(success_reply: {:ok, %{"applied" => true}}, test_pid: self())
+
+      uri = "file:/some_file.ex"
+
+      text = """
+      defmodule BasicEx do
+        def add(a) do
+          require Logger
+
+          Logger.info(to_string(add_num(a, 12)))
+        end
+
+        def add_num(a, num), do: a + num
+      end
+      """
+
+      assert {:ok, nil} =
+               ManipulatePipes.execute(
+                 ["toPipe", uri, 4, 17],
+                 %Server{
+                   source_files: %{
+                     uri => %SourceFile{
+                       text: text
+                     }
+                   }
+                 }
+               )
+
+      assert_receive {:request, "workspace/applyEdit", params}
+
+      expected_range = %{
+        "end" => %{"character" => 41, "line" => 4},
+        "start" => %{"character" => 16, "line" => 4}
+      }
+
+      expected_substitution = "to_string(a |> add_num(12))"
```

Comment:
This one might bring up some discussion. The way that our macro-walking function is implemented, the code infers that the innermost function call is the one that should be piped.

Does this need to be reworked so the outermost one is the piped call?

Reply:
I guess that would be preferred. On the other hand this is an ambiguous case.

Reply:
@lukaszsamson I've fixed this in the latest commit. The issue was that there was a `and t != []` clause which disallowed 1-arg functions from being piped. No other tests broke, so that's a good sign, I guess